### PR TITLE
Introduce Core Message File and Move Connection Failure Message Generation

### PR DESCRIPTION
**Pull request by stefaus**
This PR moves Message Generation for Connection Failures to the Core and **introduces a Message ResourceBundle File in the Core**.

---

#### [CORE] Create an Initial Core Messages File
Add an inital core messages file and add the resource export to build
config.

#### [CORE][E][I] Unify Connect ErrorMessage Generation
Eclipse and IntelliJ use nearly the same logic to generate Error
Messages for Connection Failures. This Unification aligns these messages
and is a preparation to keep XMPPException in the core.

#### [CORE][E][I] Return ErrorMessage on Connect Fail
Instead of returning the Exception on Connection Failure, return a
message with the error.

#### [CORE] Move Connect Fail Messages to Property File

### CODE REVIEW

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- core/src/saros/ui/CoreMessages.java  1
- core/src/saros/util/MessageUtils.java  6
- core/src/saros/communication/connection/ConnectionHandler.java  7
         

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/eventhandler/ConnectingFailureHandler.java  -8
+ intellij/src/saros/intellij/ui/eventhandler/ConnectingFailureHandler.java  -13
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5579204&bid=18289913)

File: core/src/saros/communication/connection/ConnectionHandler.java [hunk 1]
```diff
@@ -177,6 +180,73 @@ public void setCallback(IConnectingFailureCallback callback) {
...
+    // as of Smack 3.3.1 this is always null for connection attempts
+    // Throwable cause = e.getWrappedThrowable();
+
+    XMPPError error = xmppException.getXMPPError();
+
+    if (error != null && error.getCode() == 504)
+      return "The XMPP server "
+          + account.getDomain()
+          + " could not be found. Make sure that you are connected to the internet and that you entered the domain part of your JID correctly.\n\n"
+          // TODO re-add when settings are implemented
```
Comment (m273d15) [inline core/src/saros/communication/connection/ConnectionHandler.java]: Do you mean "when settings are implemented in IntelliJ"?

File: core/src/saros/communication/connection/ConnectionHandler.java [hunk 2]
```diff
@@ -177,6 +180,73 @@ public void setCallback(IConnectingFailureCallback callback) {
     this.callback = callback;
   }
 
+  // TODO make private, keep XMPPException in core, move messages to CoreMessages
+  public static String generateHumanReadableErrorMessage(
+      XMPPAccount account, XMPPException xmppException) {
+
+    // as of Smack 3.3.1 this is always null for connection attempts
+    // Throwable cause = e.getWrappedThrowable();
```
Comment (m273d15) [inline core/src/saros/communication/connection/ConnectionHandler.java]: Why do we keep this code?

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- core/src/saros/communication/connection/ConnectionHandler.java  7
- core/src/saros/ui/CoreMessages.java  1
- core/src/saros/util/MessageUtils.java  6
         

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/eventhandler/ConnectingFailureHandler.java  -8
+ intellij/src/saros/intellij/ui/eventhandler/ConnectingFailureHandler.java  -13
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5579204&bid=18289913)

File: core/src/saros/communication/connection/ConnectionHandler.java (hunk 1 above)
Comment (stefaus) [reply core/src/saros/communication/connection/ConnectionHandler.java]: I more or less ignored this message, but added it in a slightly modified way as IntelliJ allows this setting on Account adding and would use it when modified via eclipse.

File: core/src/saros/communication/connection/ConnectionHandler.java (hunk 2 above)
Comment (stefaus) [reply core/src/saros/communication/connection/ConnectionHandler.java]: It is removed in a following commit. (I think the one you looked at was the one where I just copied the IntelliJ Method).

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- core/src/saros/ui/CoreMessages.java  1
- core/src/saros/communication/connection/ConnectionHandler.java  7
- core/src/saros/util/MessageUtils.java  6
         

Complexity decreasing per file
==============================
+ intellij/src/saros/intellij/ui/eventhandler/ConnectingFailureHandler.java  -13
+ eclipse/src/saros/ui/eventhandler/ConnectingFailureHandler.java  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5579204&bid=18289913)

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- core/src/saros/ui/CoreMessages.java  1
- core/src/saros/util/MessageUtils.java  6
- core/src/saros/communication/connection/ConnectionHandler.java  7
         

Complexity decreasing per file
==============================
+ intellij/src/saros/intellij/ui/eventhandler/ConnectingFailureHandler.java  -13
+ eclipse/src/saros/ui/eventhandler/ConnectingFailureHandler.java  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5579204&bid=18289913)

**Pull request merged**